### PR TITLE
Restart Game UI, Client API Refactor, Bug Fixes

### LINKED ISSUE #21: Stop the text box from expanding for large text blocks

### LINKED ISSUE #22: Fix light mode on devices in device-level dark mode
Light mode is ~~messed~~ up! Pretty sure it's because my device is sometimes in dark mode (thanks to the auto macOS appearance).

Screen Shot 2023-10-16 at 20 15 57
Screen Shot 2023-10-16 at 20 16 42

### PR DESCRIPTION
**Pull request by WilliamHCarter**
- Fixed OS Dark mode issues for Mac
- Fixed Text expanding beyond ASCII width
- Re-added strict mode
- Added 'Play Again' Button on `gameOver`
- Added `API.ts` file to better separate API Handling/Processing from UI state.

### CODE REVIEW

File: client/src/components/AsciiBanner.tsx [hunk 1]
```diff
@@ -1,78 +1,101 @@
...
+  };
+
+  useEffect(() => {
+    updateWidth();
+    window.addEventListener("resize", updateWidth);
+
+    return () => {
+      window.removeEventListener("resize", updateWidth);
+    };
+  }, []);
```
Comment (amchugh) [inline client/src/components/AsciiBanner.tsx:20]: Nice.

File: client/src/components/InputField.tsx [hunk 2]
```diff
@@ -4,9 +4,16 @@ import { SendHorizontal, Loader2 } from "lucide-react";
 type InputFieldProps = {
   onSend: (message: string) => void;
   disabled?: boolean;
+  gameOver?: boolean;
+  onRestart?: () => void;
```
Comment (amchugh) [inline client/src/components/InputField.tsx:8]: Is that a type decoration? Never seen that before! I'm guessing it means `this is a function that takes no arguments and returns void`. Pretty cool.

File: client/src/components/ResponseHandler.tsx [hunk 3]
```diff
@@ -2,106 +2,91 @@ import { useEffect, useState } from "react";
...
-      const data = await response.json();
-      console.log(data);
-      return data;
-    }
-    return Error("Error: Server Request Failed.");
-  };
-
-  const handleUserInput = async (userInput: string) => {
-    if (gameOver) return;
+  const validateOption = (lastMessage: any, userInput: string): boolean => {
```
Comment (amchugh) [inline client/src/components/ResponseHandler.tsx:46]: I like that you broke it up a little more.

File: client/src/main.tsx [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
-import App from './App.tsx'
-import './index.css'
+import ReactDOM from "react-dom/client";
+import App from "./App.tsx";
+import "./index.css";
+import React from "react";
 
-ReactDOM.createRoot(document.getElementById('root')!).render(
+ReactDOM.createRoot(document.getElementById("root")!).render(
+  <React.StrictMode>
```
Comment (amchugh) [inline client/src/main.tsx:7]: 😎

File: client/tailwind.config.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ export default {
       colors : {
         neutral: "#f4f4f4",
         dneutral: "#1f1f1f",
-        lbg: "ffffff",
+        lbg: "#ffffff",
```
Comment (amchugh) [inline client/tailwind.config.js:15]: Lol. Good catch.

File: client/src/API.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  const command = castCommand(data.response);
+  const styles = extractTextStyles(command);
+  if (data.message && typeof data.message === "string") {
+    setConversation((prev) => [...prev, data.message]);
+    setGameID(data.game_id);
+    setStyles(styles);
+  }
+};
+
+export async function playGame(
```
Comment (amchugh) [inline client/src/API.ts:34]: This doesn't play the whole game, just the server/game's "turn"? We should come up with a name for this... how about [ply](https://en.wikipedia.org/wiki/Ply_(game_theory))?

**Review by amchugh — APPROVED** [review]

File: client/src/components/InputField.tsx (hunk 2 above)
Comment (WilliamHCarter) [reply client/src/components/InputField.tsx:8]: yep!

File: client/tailwind.config.js (hunk 5 above)
Comment (WilliamHCarter) [reply client/tailwind.config.js:15]: A classic

**Pull request merged**